Source/gui/SNTBinaryMessageWindowView.swift

```diff
@@ -238,12 +238,27 @@ struct SNTBinaryMessageWindowView: View {
   let event: SNTStoredEvent?
   let customMsg: NSString?
   let customURL: NSString?
-  let uiStateCallback: ((Bool) -> Void)?
+  let uiStateCallback: ((TimeInterval) -> Void)?
 
   let c = SNTConfigurator()
 
   @State public var preventFutureNotifications = false
 
+  let preventNotificationPeriods: [TimeInterval] = [86400, 604800, 2592000]
+  @State public var preventFutureNotificationPeriod: TimeInterval = 86400
+  let dateFormatter = DateComponentsFormatter()
+
+  init(window: NSWindow, event: SNTStoredEvent?, customMsg: NSString?, customURL: NSString?, uiStateCallback: ((TimeInterval) -> Void)?) {
+    self.window = window
+    self.event = event
+    self.customMsg = customMsg
+    self.customURL = customURL
+    self.uiStateCallback = uiStateCallback
+
+    dateFormatter.unitsStyle = .spellOut
+    dateFormatter.allowedUnits = [.hour, .day, .month]
```

Comment:
consider replacing `.hour` with `.weekOfMonth`. With the time interval change above, my local tests now show "one day, one week, one month". I think it looks a bit nicer but debatable and not a blocker if you prefer the current days-only output.

Reply:
Interesting, I didn't realize `.weekOfMonth` would do that. That's much nicer.